### ISSUE TITLE
Lacks support for sqlalchemy.orm.scoped_session.

### ISSUE THREAD

**Issue by KaranSharma-gawds** [HTTP-APIs/hydrus#99]
### I'm submitting a 
- [ ] bug report.

### Current Behaviour:

Lacks support for sqlalchemy.orm.scoped_session.
Currently the code in set_session() checks only for ```sqlalchemy.orm.session.Session```, while flask also supports making of other types of sessions.

### Expected Behaviour:

```scoped_session``` in sqlalchemy.orm will be supported

### Steps to repoduce:

Import scoped_session.
```python
from sqlalchemy.orm import scoped_session.
```
Replace the current session-creation code with the following:
```python
session = scoped_session(sessionmaker(autocommit=False, autoflush=False, bind=engine))
```
### Do you want to work on this issue ?

Yes.

**Comment by xadahiya** [comment]
I think using `scoped_session` instead of `session` is a good idea.
Those who are not familiar with `scoped_session`, please read [this](http://docs.sqlalchemy.org/en/latest/orm/contextual.html).

@Mec-iS  @chrizandr  what do you think?

**Comment by KaranSharma-gawds** [comment]
I was thinking of reducing the dependency on the kind of session object being used altogether rather than just switching it to another, as the issue of compatibility will still remain even if we change it from one to another. If some API is not using the session object included in here, then they will have to face this issue otherwise.

**Comment by Mec-iS** [comment]
So in practice we should use `scoped_session` in our code and check for the existence of different classes of sessions? Using `scoped_session` makes sense as we may want the session to have the same lifetime as the request.

**Comment by xadahiya** [comment]
Exactly!

**Comment by alohamora** [comment]
Can I work on this issue?
## My Implementation
Check for both types of session in `utils.py`:
```python
if not isinstance(DB_SESSION,scoped_session) and not isinstance(DB_SESSION,Session):
```
Use scoped_session for thread-managed registry of database session in `main.py`:
```python
session = scoped_session(sessionmaker(autocommit=False, autoflush=False, bind=engine))

**Comment by Mec-iS** [comment]
If you have a fix that pass all tests, you can open a PR and ask for a review.

Read the guidelines here: https://github.com/HTTP-APIs/hydrus/wiki/Starting-Material

**Comment by xadahiya** [comment]
Fixed in #112

**Comment by KaranSharma-gawds** [comment]
@Mec-iS @xadahiya please review the PR #162 .
